spec/client/ics-028-wasm-client/README.md

```diff
@@ -0,0 +1,451 @@
+---
+ics: 28
+title: WASM Client
+stage: draft
+category: IBC/TAO
+kind: instantiation
+implements: 2
+author: Parth Desai <parth@chorus.one>
+created: 2020-10-13
+modified: 2020-10-13
+---
+
+## Synopsis
+
+This specification document describes an interface to a client (verification algorithm) stored as a wasm bytecode for a blockchain.
+
+### Motivation
+
+WASM based light clients are decoupled from SDK source code, which enables one to upgrade existing light client or add support for other blockchain without modifying SDK source code.
+
+### Definitions
+
+Functions & terms are as defined in [ICS 2](../ics-002-client-semantics).
+
+`currentTimestamp` is as defined in [ICS 24](../ics-024-host-requirements).
+
+`WASM Client Code` refers to wasm bytecode stored in the client store, which provides target blockchain specific implementation of [ICS 2](../ics-002-client-semantics).
+
+`WASM Client` refers to a particular instance of `WASM Client Code` defined as a tuple `(WASM Client Code, ClientID)`.
+
+`WASM VM` refers to a virtual machine capable of executing valid wasm bytecode.
+
+
+### Desired Properties
+
+This specification must satisfy the client interface defined in ICS 2.
+
+## Technical Specification
+
+This specification depends on correctness of the `WASM Client code` in context of consensus algorithm of its target `blockchain`, as well as correct instantiation of `WASM Client`.
+
+### Client state
+
+The wasm client state tracks location of the wasm bytecode via `codeId`. Binary data represented by `data` field is opaque and only interpreted by the WASM Client Code. `type` represents client type.
```

Comment:
please capitalise WASM consistently (or not, but pick one)

Reply:
I believe "Wasm" (not WASM or wasm) is [the preferred abbreviation](https://webassembly.org/)

Reply:
Then let's use that consistently :)

Reply:
"WASM" has been replaced with "Wasm" across the ICS. I think we can resolve this thread :)